engine/src/cubos/engine/renderer/plugin.cpp

```diff
@@ -99,21 +100,59 @@ static void framePointLights(Write<RendererFrame> frame, Query<Read<PointLight>,
     }
 }
 
-static void draw(Write<Renderer> renderer, Read<ActiveCamera> activeCamera, Write<RendererFrame> frame,
+static void frameEnvironment(Write<RendererFrame> frame, Read<RendererEnvironment> env)
+{
+    frame->ambient(env->ambient);
+    frame->skyGradient(env->skyGradient[0], env->skyGradient[1]);
+}
+
+static void draw(Write<Renderer> renderer, Read<ActiveCameras> activeCameras, Write<RendererFrame> frame,
                  Query<Read<LocalToWorld>, Read<Camera>> query)
 {
-    cubos::core::gl::Camera glCamera;
+    cubos::core::gl::Camera cameras[2]{};
+    int cameraCount = 0;
+    glm::ivec2 size = (*renderer)->size();
+
+    for (int i = 0; i < 2; ++i)
+    {
+        if (activeCameras->entities[i].isNull())
+        {
+            continue;
+        }
+
+        if (auto components = query[activeCameras->entities[i]])
+        {
+            auto [localToWorld, camera] = *components;
+            cameras[cameraCount].fovY = camera->fovY;
+            cameras[cameraCount].zNear = camera->zNear;
+            cameras[cameraCount].zFar = camera->zFar;
+            cameras[cameraCount].view = glm::inverse(localToWorld->mat);
+            cameraCount += 1;
+        }
+    }
+
+    if (cameraCount == 0)
+    {
+        CUBOS_WARN("No active camera set - renderer skipping frame");
+    }
+    else if (cameraCount == 1)
+    {
+        cameras[0].viewportPosition = {0, 0};
+        cameras[0].viewportSize = size;
+    }
+    else if (cameraCount == 2)
+    {
+        cameras[0].viewportPosition = {0, 0};
+        cameras[0].viewportSize = {size.x / 2, size.y};
+        cameras[1].viewportPosition = {size.x / 2, 0};
+        cameras[1].viewportSize = {size.x / 2, size.y};
+    }
```

Comment:
Not a fan of this, but we can make it procedural once we have more than two cameras